multiwallet.go

```diff
@@ -0,0 +1,428 @@
+
+	exists, err := mw.WalletNameExists(walletName)
+	if err != nil {
+		return nil, err
+	} else if exists {
+		return nil, errors.New(ErrWalletNotLoaded)
+	}
+
+	w := &Wallet{
+		Name:               walletName,
+		DiscoveredAccounts: true,
+	}
+
+	err = mw.db.Save(w)
+	if err != nil {
+		return nil, err
+	}
+
+	homeDir := filepath.Join(mw.rootDir, strconv.Itoa(w.ID))
+	os.MkdirAll(homeDir, os.ModePerm) // create wallet dir
+
+	w.DataDir = homeDir
+	err = mw.db.Save(w) // update database with complete wallet information
+	if err != nil {
+		return nil, err
+	}
+
+	// delete from database if not created successfully
+	defer func() {
+		if err != nil {
+			mw.db.DeleteStruct(w)
+		}
+	}()
+
+	w.walletDbDriver = mw.dbDriver
+	w.netType = mw.activeNet.Name
+	libWallet, err := NewLibWallet(w)
+	if err != nil {
+		return nil, err
+	}
+
+	mw.wallets[w.ID] = libWallet
+
+	err = libWallet.CreateWatchingOnlyWallet(wallet.InsecurePubPassphrase, extendedPublicKey)
+	if err != nil {
+		return nil, err
+	}
+
+	go mw.listenForTransactions(libWallet)
+
+	return libWallet, nil
+}
+
+func (mw *MultiWallet) RestoreWallet(seedMnemonic, privatePassphrase string, spendingPassphraseType int32) (*LibWallet, error) {
+	if mw.activeSyncData != nil {
+		return nil, errors.New(ErrSyncAlreadyInProgress)
+	}
+
+	properties := &Wallet{
+		SpendingPassphraseType: spendingPassphraseType,
+		DiscoveredAccounts:     false,
+	}
+
+	return mw.createWallet(properties, seedMnemonic, privatePassphrase)
+}
+
+func (mw *MultiWallet) createWallet(w *Wallet, seedMnemonic, privatePassphrase string) (*LibWallet, error) {
+
+	err := mw.db.Save(w)
+	if err != nil {
+		return nil, err
+	}
+
+	walletName := "wallet-" + strconv.Itoa(w.ID) // wallet-#
+
+	homeDir := filepath.Join(mw.rootDir, strconv.Itoa(w.ID))
+	os.MkdirAll(homeDir, os.ModePerm) // create wallet dir
+
+	// update database wallet data dir
+	w.DataDir = homeDir
+	w.Name = walletName
+	err = mw.db.Save(w) // update database with complete wallet information
+	if err != nil {
+		return nil, err
+	}
+
+	// delete from database if not created successfully
+	defer func() {
+		if err != nil {
+			mw.db.DeleteStruct(w)
+		}
+	}()
+
+	w.walletDbDriver = mw.dbDriver
+	w.netType = mw.activeNet.Name
+	libWallet, err := NewLibWallet(w)
+	if err != nil {
+		return nil, err
+	}
+
+	libWallet.wallet = w
+	mw.wallets[w.ID] = libWallet
+
+	err = libWallet.CreateWallet(privatePassphrase, seedMnemonic)
+	if err != nil {
+		return nil, err
+	}
+
+	go mw.listenForTransactions(libWallet)
+
+	return libWallet, nil
+}
+
+func (mw *MultiWallet) WalletNameExists(walletName string) (bool, error) {
+
+	if strings.HasPrefix(walletName, "wallet-") {
+		return false, errors.E(ErrReservedWalletName)
+	}
+
+	err := mw.db.One("Name", walletName, &Wallet{})
+	if err == nil {
+		return true, nil
+	} else if err != storm.ErrNotFound {
+		return false, err
+	}
+
+	return false, nil
+}
+
+func (mw *MultiWallet) GetWallet(walletID int) *LibWallet {
```

Comment:
As mentioned above, getters should not require a `Get` prefix. Also, this doesn't really return a wallet but a libwallet, unless we want to revisit renaming `LibWallet` to `Wallet`.

```suggestion
func (mw *MultiWallet) LibWallet(walletID int) *LibWallet {
```


I should add here, that it would be a very fine idea to rename `LibWallet` to `Wallet`, so instead of having 2 structs (i.e. `Wallet` that extends `wallet.Wallet` and `LibWallet` that holds a `wallet *Wallet` property), we could have just one struct `Wallet` that extends `wallet.Wallet` with additional properties and methods.

Reply:
This (https://github.com/raedahgroup/dcrlibwallet/pull/57/files#r344599795) was practically ignored. Let's discuss this.